### DISCUSSION TITLE
RAM gets full?

### DISCUSSION THREAD

**Discussion by DogitoErgoSum** [pyg-team/pytorch_geometric#4545] [Q&A] [answered]
Why does my ram gets full after a few hundreds iterations? the model is on cuda and so as the loaded graph

![image](https://user-images.githubusercontent.com/51344466/165408970-8eebba91-7452-42fe-804b-fd8517bcabe3.png)

**Comment by DogitoErgoSum** [comment]
My model should be on graphic card memory right?
![image](https://user-images.githubusercontent.com/51344466/165409915-e35e31f0-b657-45b8-b3f2-53b53727f8ac.png)

**Comment by rusty1s** [comment, marked as answer]
The issue is that you are never releasing the computation graph.
```python
loss_total = loss_total + float(loss)
```
should fix this.
